src/main/java/com/Elessar/app/client/MyClientServer.java

```diff
@@ -38,7 +39,7 @@ public void run() {
         try {
             final HttpServer server = HttpServer.create(new InetSocketAddress(serverName, port), 0);
             server.createContext("/p2pMessage", new P2PMsgHandler(messageQueue, metricManager));
-            server.setExecutor(null);
+            server.setExecutor(Executors.newCachedThreadPool());
```

Comment:
For now this is probably ok, but I want to point out that our ClientServer and MyServer will have different expectations of the thread pools:
- for MyServer, the goal is high throughput so in general we want to have high thread count (usually same as the number of physical cores available on the machine)
- for ClientServer, we probably want to have at most 1 or 2 threads to not overload the machine. And this should be enough for both pull/push mode.

Remember that in reality the server is running on a dedicated machine on which all system resources (CPU, memory, disk/network IO) can be exclusively used by the server. 
While on client side, there will be all different applications/programs running and in general you don't want to have one application monopolying all system resources. For example, when you are using wechat you probably still want to play music at the same time.